packages/device_info_plus/device_info_plus/example/integration_test/device_info_plus_test.dart

```diff
@@ -65,4 +66,121 @@ void main() {
       }
     }
   });
+
+  testWidgets('iOS', (WidgetTester tester) async {
+    if (Platform.isIOS) {
+      expect(iosInfo.name, isNotNull);
+      expect(iosInfo.systemName, equals('iOS'));
+      expect(iosInfo.systemVersion, isNotNull);
+      expect(iosInfo.model, equals('iPhone'));
+      expect(iosInfo.localizedModel, equals('iPhone'));
+
+      expect(iosInfo.identifierForVendor, isNotNull);
+      expect(iosInfo.isPhysicalDevice, isNotNull);
+
+      expect(iosInfo.utsname.sysname, equals('Darwin'));
+      expect(iosInfo.utsname.nodename, isNotNull);
+      expect(iosInfo.utsname.release, isNotNull);
+      expect(iosInfo.utsname.version, isNotNull);
+      expect(iosInfo.utsname.machine, isNotNull);
+    }
+  });
+
+  testWidgets('android', (WidgetTester tester) async {
+    if (Platform.isAndroid) {
```

Comment:
Instead of having an if condition here, you can set the `skip:` parameter of the `testWidgets` to `skip: !Platform.isAndroid` so the test will only run when running on an Android device. I think this is worth trying.

Reply:
I found this [usage](https://github.com/dart-lang/test/issues/545#:~:text=failure%27)%3B%0A%20%20%20%20%7D-,%2C%20skip%3A%20false,-)%3B%0A%20%20%7D%2C%20skip%3A) through google, not sure if it is what you are describing.
If so, I'm curious what the rationale for using this method is? Because I'm using Android studio's automation tools to run a command similar to the following.
`/Users/cheverjohn/SE/envs/flutter/flutter3/bin/flutter --no-color test --machine --start-paused --name "Can get non\-null device model(\s*\( variant: . *\))? $" -d 24c0f2776eb8a90283ef5f22dfad19984fb7615e`
integration_test/device_info_plus_test.dart

Seems simpler?

Reply:
Oh, I see. 
Maybe This is an unnecessary judgment statement.

Reply:
I am not sure if we are talking about the same.

The `skip` parameter from the `testWidgets` allows you to disable a test from running given a certain condition. The problem is that if you run this test on a non-Android machine it will always pass, even if it did nothing, giving a false positive. It would be better if instead the test was skipped completely.